Added a CONTRIBUTING.md to Guide New Contributors (#680)
CopilotKit/CopilotKit: CONTRIBUTING.md

## Step 1: Make a fork

Fork the CopilotKit repository to your GitHub organization. This means that you'll have a copy of the repository under _your-GitHub-username/repository-name_.

## Step 2: Clone the repository to your local machine

```
git clone https://github.com/{your-GitHub-username}/CopilotKit

```

## Step 3: Prepare the development environment

Set up and run the development environment on your local machine:

### 1)Install Prerequisites
- Node.js 20.x or later
- pnpm v9.x installed globally (npm i -g pnpm@^9)
- Turborepo v2.x installed globally (npm i -g turbo@2)

### 2)Install Dependencies
Install the dependencies using pnpm:

```jsx
pnpm install
```
### 3)Build Packages
To make sure everything works, let’s build all packages once:

```jsx
turbo run build
```

## Step 4: Create a branch

Create a new branch for your changes.
In order to keep branch names uniform and easy-to-understand, please use the following conventions for branch naming.
Generally speaking, it is a good idea to add a group/type prefix to a branch.
Here is a list of good examples:

> - for docs change : docs/{ISSUE_NUMBER}-{CUSTOM_NAME}

Reviewer: ```suggestion
- for docs change : docs/<ISSUE_NUMBER>-<CUSTOM_NAME>
```
same with others

- for new features : feat/{ISSUE_NUMBER}-{CUSTOM_NAME}
- for bug fixes : fix/{ISSUE_NUMBER}-{CUSTOM_NAME}

```jsx

> git checkout -b branch-name-here

Reviewer: ```suggestion
git checkout -b <new-branch-name-here>
```

```

## Step 5: Make your changes

Now that everything is set up and works as expected, you can get start developing or update the code with your bug fix or new feature.

```jsx

> # Start all packages in development mode

Reviewer: ```suggestion
# To start all packages in development mode
```

turbo run dev
 
# Start a specific package in development mode
turbo run dev --filter="@copilotkit/package-name"
```

## Step 6: Add the changes that are ready to be committed

Stage the changes that are ready to be committed:

```jsx
git add .
```

## Step 7: Commit the changes (Git)

Commit the changes with a short message. (See below for more details on how we structure our commit messages)

```jsx
git commit -m "<type>(<package>): <subject>"
```

## Step 8: Push the changes to the remote repository

Push the changes to the remote repository using:

```jsx

> git push origin branch-name-here

Reviewer: ```suggestion
git push origin <branch-name-here>
```

# Contributing to CopilotKit

⭐ Thank you for your interest in contributing!!

Here’s how you can contribute to this repository

## How can I contribute?

> 

Reviewer: I do think you have answered this in later section then why redirecting to discord here.
Instead can add this as faq and place this in later section

Ready to contribute but seeking guidance, we have several avenues to assist you. Explore the upcoming segment for clarity on the kind of contributions we appreciate and how to jump in. Reach out to us directly on [Discord](https://discord.gg/6dffbvGU3D) for immediate assistance! Alternatively, you're welcome to raise an issue and one of our dedicated maintainers will promptly steer you in the right direction!

## Found a bug?

If you find a bug in the source code, you can help us by [submitting an issue](https://github.com/CopilotKit/CopilotKit/issues/new?assignees=&labels=bug&projects=&template=bug_report.yaml) to our GitHub Repository. Even better, you can submit a Pull Request with a fix.

## Missing a feature?

So, you've got an awesome feature in mind? Throw it over to us by [creating an issue](https://github.com/CopilotKit/CopilotKit/issues/new?assignees=&labels=feature-request&projects=&template=feature_request.yaml) on our GitHub Repo.

If you don't feel ready to make a code contribution yet, no problem! You can also check out the [documentation issues](https://github.com/CopilotKit/CopilotKit/issues?q=is%3Aopen+is%3Aissue+label%3Adocumentation).

> # How do I make a code contribution?

Reviewer: isn't this is the answer to previous question How can I contribute you can redesign this part a bit i think

## Good first issues

Are you new to open source contribution? Wondering how contributions work in our project? Here's a quick rundown.

Find an issue that you're interested in addressing, or a feature that you'd like to add.
You can use [this view](https://github.com/CopilotKit/CopilotKit/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) which helps new contributors find easy gateways into our project.

## Step 1: Make a fork

Fork the CopilotKit repository to your GitHub organization. This means that you'll have a copy of the repository under _your-GitHub-username/repository-name_.

![Group 3](https://github.com/user-attachments/assets/7c2b8d15-87cf-4cc7-be86-5fadaebfad0b)

## Step 2: Clone the repository to your local machine

```
git clone https://github.com/<your-GitHub-username>/CopilotKit

```

![Group 4](https://github.com/user-attachments/assets/e3e78b2b-eead-463b-858b-8d40e4cb18e9)

## Step 3: Prepare the development environment

> ### 1)Install Prerequisites

Reviewer: ```suggestion
### 1) Install Prerequisites
```

- Node.js 20.x or later
- pnpm v9.x installed globally (npm i -g pnpm@^9)
- Turborepo v2.x installed globally (npm i -g turbo@2)

> ### 2)Install Dependencies

Reviewer: ```suggestion
### 2) Install Dependencies
```
same with the rest

```

## Step 4: Create a branch

Create a new branch for your changes.
In order to keep branch names uniform and easy-to-understand, please use the following conventions for branch naming.
Generally speaking, it is a good idea to add a group/type prefix to a branch.
Here is a list of good examples:

- for docs change : docs/<ISSUE_NUMBER>-<CUSTOM_NAME>
- for new features : feat/<ISSUE_NUMBER>-<CUSTOM_NAME>
- for bug fixes : fix/<ISSUE_NUMBER>-<CUSTOM_NAME>

```jsx
git checkout -b <new-branch-name-here>
```

## Step 5: Make your changes

Now that everything is set up and works as expected, you can get start developing or update the code with your bug fix or new feature.

```jsx
# To start all packages in development mode
turbo run dev
 
# Start a specific package in development mode
turbo run dev --filter="@copilotkit/package-name"
```

## Step 6: Add the changes that are ready to be committed

Stage the changes that are ready to be committed:

```jsx
git add .
```

## Step 7: Commit the changes (Git)

Commit the changes with a short message. (See below for more details on how we structure our commit messages)

```jsx
git commit -m "<type>(<package>): <subject>"
```

> ## Step 8: Push the changes to the remote repository

Reviewer: I think you should also mention pulling the latest changes from remote before pushing to avoid conflicts

Push the changes to the remote repository using:

```jsx
git push origin <branch-name-here>
```

## Step 9: Create Pull Request

In GitHub, do the following to submit a pull request to the upstream repository:

> 1.  Give the pull request a title and a short description of the changes made. Include also the issue or bug number associated with your change. Explain the changes that you made, any issues you think exist with the pull request you made, and any questions you have for the maintainer.

Reviewer: ```suggestion
1. Open a pull request with a title and a brief description of the changes made. Be sure to tag the associated issue or bug using #<Issue_No>. In the description, explain the changes you made, any issues you think might exist with the pull request, and any questions you have for the maintainer.
```
Plz fix the grammatical errors in your writing using some online tool.

Remember, it's okay if your pull request is not perfect (no pull request ever is). The reviewer will be able to help you fix any problems and improve it!

2.  Wait for the pull request to be reviewed by a maintainer.

3.  Make changes to the pull request if the reviewing maintainer recommends them.

Celebrate your success after your pull request is merged :-)

> ## Git Commit Messages

Reviewer: I think you should add this as well into above section inside 'Commit the changes'
